integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java

```diff
@@ -0,0 +1,502 @@
+        "tunneling", "voyager.tunneling.template.yaml");
+    Path targetVoyagerHttpFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcVoyagerHttpFile.toString(), 
+        "voyager.tunneling.yaml", templateMap));
+    logger.info("Generated Voyager Http Tunneling file {0}", targetVoyagerHttpFile);
+
+    StringBuffer deployIngress = new StringBuffer("kubectl apply -f ");
+    deployIngress.append(Paths.get(RESULTS_ROOT, "voyager.tunneling.yaml"));
+
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(deployIngress), true));
+
+    logger.info("kubectl apply returned {0}", result.toString());
+    checkServiceExists("voyager-voyager-tunneling", domainNamespace);
+
+    int httpTunnelingPort =
+        getServiceNodePort(domainNamespace, "voyager-voyager-tunneling");
+    assertTrue(httpTunnelingPort != -1,
+        "Could not get the HttpTunnelingPort service node port");
+    logger.info("HttpTunnelingPort for Voyager {0}", httpTunnelingPort);
+
+    StringBuffer httpUrl = new StringBuffer("http://");
+    httpUrl.append(K8S_NODEPORT_HOST + ":" + httpTunnelingPort);
+
+    StringBuffer javaCmd = new StringBuffer("java -cp ");
+    javaCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar"));
+    javaCmd.append(":");
+    javaCmd.append(Paths.get(RESULTS_ROOT));
+    javaCmd.append(" JmsTestClient ");
+    javaCmd.append(httpUrl);
+    logger.info("java command to be run {0}", javaCmd.toString());
+
+    // Note it takes a couples of iterations before the client success
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Wait for Http JMS Client to access WLS "
+                    + "(elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(runJmsClient(new String(javaCmd)));
+  }
+
+  @Order(2)
+  @Test
+  @DisplayName("Verify tls RMI access WLS through LoadBalancer tunneling port")
+  public void testExternalRmiAccessThruHttpsTunneling() {
+
+    buildClient();
+
+    // Prepare the voyager ingress file from the template file by replacing 
+    // domain namespace, domain UID, cluster service name and tls secret 
+    Map<String, String> templateMap  = new HashMap();
+    templateMap.put("DOMAIN_NS", domainNamespace);
+    templateMap.put("DOMAIN_UID", domainUid);
+    templateMap.put("CLUSTER", clusterName);
+    templateMap.put("TLS_CERT", tlsSecretName);
+
+    Path srcVoyagerHttpsFile  = Paths.get(RESOURCE_DIR,
+        "tunneling", "voyager.tls.tunneling.template.yaml");
+    Path targetVoyagerHttpsFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcVoyagerHttpsFile.toString(),
+            "voyager.tls.tunneling.yaml", templateMap));
+    logger.info("Generated Voyager Https Tunneling file {0}", targetVoyagerHttpsFile);
+
+    createCertKeyFiles(K8S_NODEPORT_HOST);
```

Comment:
add comments

Reply:
The method definition createCertKeyFiles() has the description
// Create display SSL certificate and key using openssl with SAN extension

Reply:
The method definition createCertKeyFiles() has the comment

Reply:
Added comments inside the test method